core/src/main/java/hudson/cli/InstallPluginCommand.java

```diff
@@ -104,7 +105,7 @@ protected int run() throws Exception {
 
             // is this an URL?
             try {
-                URL u = new URL(source);
+                URL u = new URI(source).toURL();
```

Comment:
neets to be handled in line 117 along with the `MalformedUrlException`

Reply:
added `e.getMessage();` under `catch(MalformedUrlException e)`

Reply:
this has the potential to throw `URISyntaxException`,  the code already handled `MalformedURLException` , it is the newly thrown `URISyntaxException` that needs to be handled on line 117 in the same way.

Reply:
replaced/added with
```
} catch (MalformedURLException | URISyntaxException e) {
                // not an URL
                e.getCause();
            }
```